dev/src/watch.ts

```diff
@@ -426,6 +433,15 @@ abstract class Watch {
      * Initializes a new stream to the backend with backoff.
      */
     const initStream = () => {
+      if (this._backoff.getRetryCount() > DEFAULT_MAX_RETRY_ATTEMPTS) {
+        closeStream(
+          new Error(
+            'Exceeded maximum number of retries before any ' +
+              'response was received.'
+          )
+        );
+        return;
+      }
       this._backoff.backoffAndWait().then(async () => {
```

Comment:
What do you think about encapsulating `retryCount` in `backoffAndWait()` and failing the Promise that `backoffAndWait()` returns?